network/server.go

```diff
@@ -759,14 +759,16 @@ func StringToAddrInfo(addr string) (*peer.AddrInfo, error) {
 }
 
 var (
-	// Regex used for matching loopback addresses (IPv4 and IPv6)
+	// Regex used for matching loopback addresses (IPv4, IPv6, DNS)
 	// This regex will match:
 	// /ip4/localhost/tcp/<port>
 	// /ip4/127.0.0.1/tcp/<port>
 	// /ip4/<any other loopback>/tcp/<port>
 	// /ip6/<any loopback>/tcp/<port>
+	// /dns/foobar.com/tcp/<port>
 	loopbackRegex = regexp.MustCompile(
-		`^\/ip4\/127(?:\.[0-9]+){0,2}\.[0-9]+\/tcp\/\d+$|^\/ip4\/localhost\/tcp\/\d+$|^\/ip6\/(?:0*\:)*?:?0*1\/tcp\/\d+$`,
+		//nolint:lll
+		`^\/ip4\/127(?:\.[0-9]+){0,2}\.[0-9]+\/tcp\/\d+$|^\/ip4\/localhost\/tcp\/\d+$|^\/ip6\/(?:0*\:)*?:?0*1\/tcp\/\d+$|^\/dns\/[^-|^/][A-Za-z0-9-]([^-|^/]?)+([\\-\\.]{1}[a-z0-9]+)*\\.[A-Za-z]{2,}(/?)$`,
```

Comment:
What do you think about extracting the last part of the regex you've reused from [here](https://github.com/0xPolygon/polygon-edge/blob/develop/helper/flags/flags.go#L55) into the `common` or `helper` packages, and just referencing it here?

We use it in multiple places now with the introduction of this PR, and future fixes should propagate to all usages.

Reply:
👍🏼 . have extracted the regex to a simple helper func.  
Resolved here https://github.com/0xPolygon/polygon-edge/pull/406/commits/63c2bd6d1cd5d8f7ab3ad0a52b84a1de8baa9a6b

Reply:
as suggested my @Kourin1996. defining a constant for now. resolved here https://github.com/0xPolygon/polygon-edge/commit/2ad796dbf3e4884a9b15f72e9fa9059d88744282